src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,31 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirm extends Popup {
+  constructor(popupSelector) {
+    super(popupSelector);
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._confirmButton = document.querySelector("#modal-delete-btn");
+    this._saveButton = document.querySelector(".modal__button");
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    document.addEventListener("keydown", this._handleEscClose);
+    this._popupForm.addEventListener("submit", (e) => {
+      e.preventDefault();
+      this._handleDeleteSubmit();
+    });
+  }
+
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._confirmButton.textContent = "Saving...";
+    } else {
+      this._confirmButton.textContent = "Yes";
+    }
+  }
+
+  setSubmitAction(action) {
+    this._handleDeleteSubmit = action;
```

Comment:
Better to rename this function to something like `handleConfirm`, so this class will be able to handle any confirmation, not only the delete confirmation